### PR TITLE
arm64 md cleanup

### PR DESCRIPTION
**Pull request by sigatrev**

### CODE REVIEW

**Comment by sigatrev** [comment]
@dotnet-bot test this please

File: lib/Backend/arm64/LinearScanMD.h [hunk 1]
```diff
@@ -27,10 +27,7 @@ class LinearScanMD : public LinearScanMDShared
     void        InsertOpHelperSpillAndRestores(SList<OpHelperBlock> *opHelperBlockList);
     void        EndOfHelperBlock(uint32 helperSpilledLiveranges);
 
-    uint        UnAllocatableRegCount(Func *func) const
-                {
-                    return func->GetLocalsPointer() != RegSP ? 5 : 4; //r11(Frame Pointer),r12,sp,pc
-                }
+    uint        UnAllocatableRegCount(Func *func) const { return 5; }
```
Comment (agarwal-sandeep) [inline lib/Backend/arm64/LinearScanMD.h]: Add a comment here to explain why 5? Also don't use hardcoded constant define a descriptive macro and use that instead.

**Comment by agarwal-sandeep** [comment]
@dotnet-bot test this please

File: lib/Backend/arm64/LinearScanMD.h (hunk 1 above)
Comment (sigatrev) [reply lib/Backend/arm64/LinearScanMD.h]: 5 is the number of regs marked as RA_DONTALLOCATE in RegList.h. Will update to use a named constant.

Comment (agarwal-sandeep) [reply lib/Backend/arm64/LinearScanMD.h]: Just add a comment here to mention the same.

**Pull request merged**